packages/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -425,30 +425,25 @@ extension on ClassElement {
   Iterable<VariableElement> allFields({
     required bool hasFreezed,
     required List<ConstructorElement> freezedConstructors,
-  }) sync* {
+  }) {
     if (hasFreezed) {
-      yield* freezedConstructors.single.parameters;
+      return freezedConstructors.single.parameters;
     } else {
       final uniqueFields = <String, FieldElement>{};
 
-      for (final field in fields) {
-        if (field.getter != null && !field.getter!.isSynthetic) {
-          continue;
-        }
-        uniqueFields[field.name] ??= field;
-      }
+      final allFields = [
+        ...fields,
+        ...allSupertypes
+            .where((e) => !e.isDartCoreObject)
+            .expand((e) => e.element.fields),
+      ];
 
-      for (final supertype in allSupertypes) {
```

Comment:
This change also excludes static fields from supertypes. Do you mind adding a test for that too?